src/cmap/conn/mod.rs

```diff
@@ -490,49 +529,63 @@ impl PinnedConnectionHandle {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct LoadBalancedGeneration {
+    pub(crate) generation: u32,
+    pub(crate) service_id: ObjectId,
+}
+
+/// TODO: RUST-1454 Once we have separate types for pooled and non-pooled connections, the
+/// monitoring case and the Option<> wrapper can be dropped from this.
+#[derive(Debug, Clone, Copy)]
 pub(crate) enum ConnectionGeneration {
+    Monitoring,
     Normal(u32),
-    LoadBalanced {
-        generation: u32,
-        service_id: ObjectId,
-    },
+    LoadBalanced(Option<LoadBalancedGeneration>),
```

Comment:
This was updated to include an `Option` to model the situation where we have a load balanced connection that has not yet received its service id. In RUST-1454 we can make this more robust and remove the need for this per the comment above.

Reply:
oh right, I found that code later on but forgot to come back and get rid of this comment :) thanks